src/main/java/net/kyori/adventure/text/minimessage/transformation/inbuild/GradientTransformation.java

```diff
@@ -76,41 +72,39 @@ private GradientTransformation() {
   }
 
   @Override
-  public void load(final String name, final List<Token> args) {
+  public void load(final String name, final List<String> args) {
     super.load(name, args);
 
     if(!args.isEmpty()) {
       final List<TextColor> textColors = new ArrayList<>();
       for(int i = 0; i < args.size(); i++) {
-        final Token arg = args.get(i);
-        if(arg.type() == TokenType.STRING) {
-          // last argument? maybe this is the phase?
-          if(i == args.size() - 1) {
-            try {
-              this.phase = Float.parseFloat(arg.value());
-              if(this.phase < -1f || this.phase > 1f) {
-                throw new ParsingException(String.format("Gradient phase is out of range (%s). Must be in the range [-1.0f, 1.0f] (inclusive).", this.phase), -1);
-              }
-              if(this.phase < 0) {
-                this.negativePhase = true;
-                this.phase = 1 + this.phase;
-              }
-              break;
-            } catch(final NumberFormatException ignored) {
+        final String arg = args.get(i);
+        // last argument? maybe this is the phase?
+        if(i == args.size() - 1) {
+          try {
+            this.phase = Float.parseFloat(arg);
+            if(this.phase < -1f || this.phase > 1f) {
+              throw new ParsingException(String.format("Gradient phase is out of range (%s). Must be in the range [-1.0f, 1.0f] (inclusive).", this.phase), -1);
```

Comment:
It would be really nice to be able to pass enough context through here to be able to provide a column index in the exception